Tools/Create-Azure-Sentinel-Solution/common/createCCPConnector.ps1

```diff
@@ -0,0 +1,880 @@
+
+                if ($TemplateCounter -eq 1) {
+                    #========start:dc definition resource===========
+                    $dcDefinitionFilteredPath = $ccpItem.DCDefinitionFilePath.Replace($solutionName + "/", "").Replace($dcFolderName + "/", "")
+                    $ccpDataDefinitionFilePath = $solutionBasePath + "/" + $solutionName + "/" + $dcFolderName + "/" + $dcDefinitionFilteredPath
+                    $ccpDataDefinitionFilePath = $ccpDataDefinitionFilePath.Replace("//", "/")
+                    Write-Host "CCP DataDefinition File Path : $ccpDataDefinitionFilePath"
+                    
+                    #$fileContent = Get-Content -Raw "$ccpDataDefinitionFilePath" | Out-String | ConvertFrom-Json
+                    $fileContent = ReadFileContent -filePath $ccpDataDefinitionFilePath
+                    if ($null -eq $fileContent) {
+                        exit 1;
+                    }
+
+                    if($fileContent.type -eq "Microsoft.SecurityInsights/dataConnectorDefinitions")
+                    {
+                        Write-Host "Processing for CCP DataDefinition file path: $dcDefinitionFilteredPath"
+                        $resourceName = "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/',variables('_dataConnectorContentId$($templateKindByCounter[1])$($global:connectorCounter)'))]"
+
+                        $armResource = Get-ArmResource $resourceName $fileContent.type $fileContent.kind $fileContent.properties
+                        $armResource.type = "Microsoft.OperationalInsights/workspaces/providers/dataConnectorDefinitions"
+
+                        $hasLocationProperty = [bool]($armResource.PSobject.Properties.name -match "location")
+                        if ($hasLocationProperty) {
+                            $locationProperty = $armResource.location
+                            $placeHoldersMatched = $locationProperty | Select-String $placeHolderPatternMatches -AllMatches
+
+                            if ($placeHoldersMatched.Matches.Value.Count -gt 0) {
+                                $placeHolderName = $placeHoldersMatched.Matches.Value.replace("{{", "").replace("}}", "")
+                                $armResource.location = "[[parameters('$($placeHolderName)')]"
+                                $templateContentConnections.properties.mainTemplate = addNewParameter -templateResourceObj $templateContentConnections.properties.mainTemplate -parameterName $placeHolderName -isSecret $false
+                            }
+                        }
+
+                        $templateContentConnectorDefinition = $templateContent;
+                        $templateContentConnectorDefinition.properties.mainTemplate.resources += $armResource
+
+                        $activeResource += $armResource
+                        $activeResource += Get-MetaDataResource 1 $dataFileMetadata $solutionFileMetadata
```

Comment:
Why this hardcoded value as 1? why cant you pass the templatecounter?

Reply:
updated